File: aoj/volume100/csharp/A10016.cs
using System;

class AOJ
{
	static void Main()
	{
		while (true)
		{
			var input = Console.ReadLine().Split();
			var m = int.Parse(input[0]);
			var f = int.Parse(input[1]);
			var r = int.Parse(input[2]);
			if (m == -1 && f == -1 && r == -1)
				break;

			Console.WriteLine(Grade(m, f, r));
		}
	}

	static string Grade(int m, int f, int r)
	{
		if (m == -1 || f == -1)        return "F";
		else if (m+f >= 80)            return "A";
		else if (m+f >= 65)            return "B";
		else if (m+f >= 50)            return "C";
		else if (m+f >= 30 && r >= 50) return "C";
		else if (m+f >= 30)            return "D";
		else                           return "F";
	}
}

File: aoj/volume100/scala/A10016.scala
object Main {
  def main(args: Array[String]) {
    loop
  }

  def loop {
    val input = scala.io.StdIn.readLine split " "
    val m = input(0).toInt
    val f = input(1).toInt
    val r = input(2).toInt
    if (m == -1 && f == -1 && r == -1) {
      return
    }

    val grade = toGrade(m, f, r)
    println(grade)
    loop
  }

  def toGrade(m:Int, f:Int, r:Int):String = {
    if (m == -1 || f == -1)        "F"
    else if (m+f >= 80)            "A"
    else if (m+f >= 65)            "B"
    else if (m+f >= 50)            "C"
    else if (m+f >= 30 && r >= 50) "C"
    else if (m+f >= 30)            "D"
    else                           "F"
  }
}
